expo/expo
# [stories] implement expo stories RN app

# Why



This is another PR aimed to break up RFC #13637 into smaller chunks

The app is a react native client that can read and render the components that are generated by the stories CLI. It also includes a set of base components that the RN app and other story (future) `.stories` files can use, as well as the TestRunner component for including `test-suite` files in stories

## Demo: 


https://user-images.githubusercontent.com/40680668/130497495-117f178c-6109-4a87-a990-72a31845d7a0.mov



# How



The CLI puts the generated `stories.js` file in the `__generated__` folder of a project, which contains all the components and configuration for a given package. 

The app uses a patched metro config to read from this file and render the components in a relatively dynamic way

# Test Plan



The best way to test this would be to use a sandbox app in `expo/expo`: 
- create a `.stories` file with an exported component in new app directory
- run the expo stories cli from the app directory - ensure the files are in `__generated__`
- update the metro config to use the exported `metro-config` from `expo-stories/metro-config`
- import `<ExpoStoriesApp />`  from 'expo-stories/clients/app` and run as the root navigator

# Checklist
